homeassistant/components/fritzbox_callmonitor/base.py

```diff
@@ -0,0 +1,65 @@
+"""Base class for fritzbox_callmonitor entities."""
+import re
+
+from fritzconnection.lib.fritzphonebook import FritzPhonebook
+
+from homeassistant.util import Throttle
+
+from .const import LOGGER, MIN_TIME_PHONEBOOK_UPDATE
+
+
+class FritzBoxPhonebook:
+    """This connects to a FritzBox router and downloads its phone book."""
+
+    def __init__(self, host, port, username, password, phonebook_id, prefixes):
+        """Initialize the class."""
+        self.host = host
+        self.username = username
+        self.password = password
+        self.port = port
+        self.phonebook_id = phonebook_id
+        self.phonebook_dict = None
+        self.number_dict = None
+        self.prefixes = prefixes
+
+        # Establish a connection to the FRITZ!Box.
```

Comment:
Not sure if it's new code, but we should avoid doing I/O in constructors. Prefer a seperate `setup` or `init` method.